cgpsa

```diff
@@ -1170,39 +1188,64 @@ sub checkMessageAddHeader
           }
         }
       }
-      
+
       $rewritten_mail->finish();
       $status->finish();
       $mail->finish();
     }
   };
 
-  if ($@) 
+  if ($@)
   {
     # Something bad happened with SpamAssassin
-    
+
     debug(4, "SpamAssassin failed in ".sprintf("%.1f", getTime() - $start).
              " seconds: $@");
     $newheaders = $root_conf->{loop_prevention_header}.": Scan Failed\e";
     ${$score_ptr} = 0.0;
   }
-  
+
   # remove any double carriage returns...
-  
-  $newheaders =~ s/\n/\e/g;   # replace \n's with \e's      
+
+  $newheaders =~ s/\n/\e/g;   # replace \n's with \e's
   $newheaders =~ s/\e\e/\e/g; # remove any double \e's
-      
+
   return $newheaders;
 }
 
+# Compute the X-Junk-Score header for a given SpamAssassin status object
+# and CGPSA configuration.
+
+sub x_junk_score_header
+{
+  my ($status, $conf, $terminator) = @_;
+  our @x_junk_score_constants;
+  our @x_junk_score_thresholds;
 
-# Check the specified message with SpamAssassin, using the specified 
+  my $junk_score_count = 0;
+  foreach my $junk_threshold (@x_junk_score_thresholds)
+  {
+    if ($status->get_score() >= $junk_threshold){
+      $junk_score_count = $junk_score_count + 1;
```

Comment:
If `@x_junk_score_thresholds` is not in increasing order (by user error in config), this may produce unexpected results.

A more robust version could be to add an `else break` branch. So that the first higher threshold than the score would stop checking the thresholds.

Reply:
Nevermind, I just saw the line:

    $legit_thresholds && isFloat($threshold) && $last_threshold < $threshold;